src/core/LogSystem.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <godot_cpp/classes/engine.hpp>
+#include <godot_cpp/classes/ref_counted.hpp>
+#include <godot_cpp/core/class_db.hpp>
+#include <godot_cpp/variant/array.hpp>
+#include <godot_cpp/variant/string.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+#include <map>
+
+#ifdef DEBUG_ENABLED
+// #define DEBUG(msg, ...) LogSystem::get_instance()->debug("[debug] " + vformat(String(msg), ##__VA_ARGS__))
+#define DEBUG(msg, ...)                                                                                                \
+    UtilityFunctions::print_rich("[color=orange](debug) " + vformat(String(msg), ##__VA_ARGS__) + "[/color]")
```

Comment:
`##__VA_ARGS__` is used only in `gcc` it will be good to add `gcc` as dependency/requirement do README (with version).

Reply:
updated the code, works for me, but have no idea how about other compilers